src/assets/woodpeckerCI/woodpeckerCI.md

```diff
@@ -0,0 +1,262 @@
+
+```yml
+  woodpecker-server:
+    image: woodpeckerci/woodpecker-server:v3
+    environment:
+      - WOODPECKER_LOG_LEVEL=debug
+      - WOODPECKER_OPEN=false
+      - WOODPECKER_HOST=${WOODPECKER_HOST}
+      - WOODPECKER_GITHUB=true
+      - WOODPECKER_GITHUB_CLIENT=${WOODPECKER_GITHUB_CLIENT}
+      - WOODPECKER_GITHUB_SECRET=${WOODPECKER_GITHUB_SECRET}
+      - WOODPECKER_AGENT_SECRET=${WOODPECKER_AGENT_SECRET}
+      - WOODPECKER_ADMIN=prashant-gurung899
+      # The following variable is where we set our Woodpecker CI Config Service (Starlark Conversion Service) later
+      - WOODPECKER_CONFIG_SERVICE_ENDPOINT=http://<your-config-service-server>
+    volumes:
+      - woodpecker-server-data:/var/lib/woodpecker/
+    labels:
+      - "traefik.enable=true"
+      - "traefik.http.routers.woodpecker-secure.rule=Host(`your-ci-server`)"
+      - "traefik.http.routers.woodpecker-secure.entrypoints=websecure"
+      - "traefik.http.routers.woodpecker-secure.tls.certresolver=letsencrypt"
+      - "traefik.http.routers.woodpecker-secure.tls=true"
+      - "traefik.http.services.woodpecker-secure.loadbalancer.server.port=8000"
+      # HTTP router + redirect to HTTPS
+      - "traefik.http.routers.woodpecker-http.rule=Host(`your-ci-server`)"
+      - "traefik.http.routers.woodpecker-http.entrypoints=web"
+      - "traefik.http.routers.woodpecker-http.middlewares=redirect-to-https"
+      # Redirect middleware
+      - "traefik.http.middlewares.redirect-to-https.redirectscheme.scheme=https"
+    networks:
+      - woodpecker-net
+
+  woodpecker-agent:
+    image: woodpeckerci/woodpecker-agent:v3
+    command: agent
+    restart: always
+    depends_on:
+      - woodpecker-server
+    volumes:
+      - woodpecker-agent-config:/etc/woodpecker
+      - /var/run/docker.sock:/var/run/docker.sock
+    environment:
+      - WOODPECKER_LOG_LEVEL=debug
+      - WOODPECKER_SERVER=woodpecker-server:9000
+      - WOODPECKER_AGENT_SECRET=${WOODPECKER_AGENT_SECRET}
+    networks:
+      - woodpecker-net
+```
+
+## Authenticating with GitHub
+
+To connect GitHub with Woodpecker:
+
+1. Register a new OAuth application in your GitHub developer [settings](https://github.com/settings/applications/new).
+
+2. In your GitHub OAuth App, set the authorization callback URL to `https://<your-ci-server>/authorize`
+
+![Callback URL](https://raw.githubusercontent.com/JankariTech/blog/woodpecker-ci-starlark/src/assets/woodpeckerCI/images/oauth.png)
+
+3. Copy the generated Client ID and Secret.
+
+4. Use these values in the `WOODPECKER_GITHUB_CLIENT` and `WOODPECKER_GITHUB_SECRET` env vars.
+
+After that, you can log into the Woodpecker web UI using your GitHub account.
+
+![GitHub Login](https://raw.githubusercontent.com/JankariTech/blog/woodpecker-ci-starlark/src/assets/woodpeckerCI/images/github.png)
+
+## Enabling a Repository
+From the Woodpecker UI:
+
+- Click on Add Repository.
```

Comment:
which one do I take?
maybe have somewhere higher up a step saying to create an empty repository, or some repo with example code that can be cloned